eidangservices/utils/strict.py

```diff
@@ -0,0 +1,329 @@
+                    'Invalid request query parameters: {}'.format(
+                        invalid_args))
+
+                if self.error_callback:
+                    self.error_callback(err, req)
+                else:
+                    self.handle_error(err, req)
+
+            return func(*args, **kwargs)
+
+        # decorator ()
+
+        return decorator
+
+    # parse ()
+
+    def with_strict_args(self, schemas, locations=None):
+        """
+        Wrapper of :py:func:`parse`.
+        """
+        return functools.partial(self.parse,
+                                 schemas=schemas,
+                                 locations=locations)
+
+    # with_strict_args ()
+
+    def _get_data(self, req, as_text=True,
+                  max_content_length=settings.MAX_POST_CONTENT_LENGTH):
+        """
+        Savely reads the buffered incoming data from the client.
+
+        :param req: Request the raw data is read from
+        :type req: :py:class:`flask.Request`
+        :param bool as_text: If set to :code:`True` the return value will be a
+            decoded unicode string.
+        :param int max_content_length: Max bytes accepted
+
+        :returns: Byte string or rather unicode string, respectively. Depending
+            on the :code:`as_text` parameter.
+        """
+        if req.content_length > max_content_length:
+            err = ValidationError(
+                'Request too large: {} bytes > {} bytes '.format(
+                    req.content_length, max_content_length))
+
+            if self.error_callback:
+                self.error_callback(err, req)
+            else:
+                self.handle_error(err, req)
+
+        return req.get_data(cache=True, as_text=as_text)
+
+    # _get_data ()
+
+    def handle_error(self, error, req):
+        """
+        Called if an error occurs while parsing args. By default, just logs and
+        raises ``error``.
+        """
+        raise error
+
+    # handle_error ()
+
+    def error_handler(self, func):
+        """
+        Decorator that registers a custom error handling function. The
+        function should received the raised error, request object, and the
+        `marshmallow.Schema` instance used to parse the request. Overrides
+        the parser's ``handle_error`` method.
+
+        Example: ::
+
+            from webargs import flaskparser
+
+            parser = flaskparser.FlaskParser()
+
+
+            class CustomError(Exception):
+                pass
+
+
+            @parser.error_handler
+            def handle_error(error, req, schema):
+                raise CustomError(error.messages)
+
+        :param callable func: The error callback to register.
+        """
+        self.error_callback = func
+        return func
+
+# class KeywordParser
+
+
+# -----------------------------------------------------------------------------
+class FlaskKeywordParser(KeywordParser):
+    """
+    Flask implementation of Keywordparser.
```

Comment:
Reference the `KeywordParser` using sphinx syntax i.e. :py:class:`KeywordParser`.